### PR TITLE
[DDP] Add a README

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request adds a DDP README document to summarize the current support for torch.nn.parallel.DistributedDataParallel. An internal version of this document can also be searchable by "How to do DistributedDataParallel"

Test Plan:
[skip ci]

### CODE REVIEW

File: docs/pjrt.md [hunk 1]
```diff
@@ -141,4 +141,7 @@ PJRT_DEVICE=TPU python3 xla/test/pjrt/test_train_pjrt_imagenet.py --fake_data
 
 ## PjRt and DDP
 
-TODO add a link to DDP doc
+PjRt composes really well with [the new experimental torch.nn.parallel.DistributedDataParallel feature](./ddp.md). Just run the DDP script as usual but with `PJRT_DEVICE=TPU`. Here is a full example:
```
Comment (will-cromar) [inline docs/pjrt.md:144]: We should have a caveat here that this only works on v4 for now

File: docs/ddp.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+ddp_model = DDP(model, gradient_as_bucket_view=True)
+```
+
+5. Finally launch your model with xla specific launcher.
+
+```
+xmp.spawn(demo_fn)
+```
+
+Here we have put everything together (the example is actually taken from the[ DDP tutorial](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html)). The way you code it is pretty similar to the eager experience. Just with xla specific touches on a single device plus the above five changes to your script.
```
Comment (will-cromar) [inline docs/ddp.md:47]: micro-nit: the tutorial link has a space in it

File: docs/ddp.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+
+## Background / Motivation
+
+Customers have long requested the ability to use PyTorch’s DistributedDataParallel API with xla. And here we enable it as an experimental feature.
+
+
+## How to use DistributedDataParallel
+
+For those who switched from the PyTorch eager mode to XLA, here are all the changes you need to do to convert your eager DDP model into XLA model. We assume that you already know how to use XLA [on a single device](https://github.com/pytorch/xla/blob/master/API_GUIDE.md#running-on-a-single-xla-device).
```
Comment (will-cromar) [inline docs/ddp.md:13]: Should we use a relative link to the API_GUIDE here so it doesn't always go to master?

File: docs/ddp.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
+# How to do DistributedDataParallel
```
Comment (will-cromar) [inline docs/ddp.md:1]: nit: Should we use backticks around `DistributedDataParallel` to be consistent with the other docs? e.g. https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#comparison-between-dataparallel-and-distributeddataparallel

**Review by will-cromar — APPROVED** [review]
Overall LGTM

Can you break up these lines in the markdown code like our other docs? [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) is helpful for this

**Review by JackCaoG — APPROVED** [review]

**Comment by alanwaketan** [comment]
> Overall LGTM
> 
> Can you break up these lines in the markdown code like our other docs? [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) is helpful for this

Rewrap doesn't work for me... What's the usual column width? Let me rewrap the doc manually.

**Pull request merged**